contracts/multisig-prover/src/encoding/abi.rs

```diff
@@ -130,27 +147,37 @@ pub fn make_operators(worker_set: WorkerSet) -> Operators {
     }
 }
 
-fn make_evm_operators_with_sigs(
-    signers_with_sigs: Vec<(Signer, Option<Signature>)>,
-) -> Result<Vec<Operator>, ContractError> {
+fn make_evm_operators_with_sigs<T>(
+    signers_with_sigs: Vec<(Signer, Option<Signature<T>>)>,
+) -> Result<Vec<Operator<T>>, ContractError>
+where
+    T: Recoverability,
```

Comment:
Does this need to be generic here? I think we always want `Recoverable`?

Reply:
some of the functions where used in both contexts. I can go through them again and check